chatbot.py

```diff
@@ -30,70 +31,101 @@ def message_to_bot(H, clf, learn_response):
         return B, learn_response
     if learn_response == 3:
         location_dict["destination"] = H
-        origin, destination = location_dict["origin"], location_dict["destination"]         # noqa: E501
+        origin, destination = (
+            location_dict["origin"],
+            location_dict["destination"],
+        )
         googleMapsApiModule.direction(origin, destination)
         B = "I will certainly help you with that."
         learn_response = 0
         return B, learn_response
-    if "bye" in H.lower().split(" "):       # check in words within H
+    if "bye" in H.lower().split(" "):  # check in words within H
         B = "Bye! I'll miss you!"
-        return B, learn_response      # exit loop
+        return B, learn_response  # exit loop
     if not H:
         B = "Please say something!"
-        return B, learn_response          # empty input
+        return B, learn_response  # empty input
     # grammar parsing
     subj = set()
     obj = set()
     verb = set()
     triples, root = utilities.parse_sentence(H)
     triples = list(triples)
     for t in triples:
-        if t[0][1][:2] == 'VB':
+        if t[0][1][:2] == "VB":
             verb.add(t[0][0])
         relation = t[1]
-        if relation[-4:] == 'subj':
+        if relation[-4:] == "subj":
             subj.add(t[2][0])
-        if relation[-3:] == 'obj':
+        if relation[-3:] == "obj":
             obj.add(t[2][0])
-    logging.debug("\t"+"Subject: "+str(subj)+"\n"+"\t"+"Object: "+str(obj)+"\n"+"\t"+"Topic: "+str(root)+"\n"+"\t"+"Verb: "+str(verb))  # noqa: E501
+    logging.debug(
+        "\t"
+        + "Subject: "
+        + str(subj)
+        + "\n"
+        + "\t"
+        + "Object: "
+        + str(obj)
+        + "\n"
+        + "\t"
+        + "Topic: "
+        + str(root)
+        + "\n"
+        + "\t"
+        + "Verb: "
+        + str(verb)
+    )
     subj = list(subj)
     obj = list(obj)
     verb = list(verb)
     proper_nouns = set()
     for t in triples:
-        if t[0][1] == 'NNP':
+        if t[0][1] == "NNP":
             proper_nouns.add(t[0][0])
-        if t[2][1] == 'NNP':
+        if t[2][1] == "NNP":
             proper_nouns.add(t[2][0])
     proper_nouns == list(proper_nouns)
```

Comment:
This line needs changes. `==` needs to be `=` I guess.

Reply:
I don't think that's the case, what makes you think so?

Reply:
This statement isn't assigned. If `==` is to be used, shouldn't there be a conditional `if` statement?

Reply:
As far as I understand, by using `=`, the statement would change `set` to a `list`.

Reply:
I was looking at a different statement, the one with the modification. Will check this.

Reply:
@chttrjeankr good observation here, it seems like this line has been this way for quite some time. Since the nouns haven't been used for anything other than printing, this seems to have skipped our notice and hasn't failed either. Yes, it can be changed to `=`.